app/experimenter/static/js/tests/PrefRolloutForm.test.js

```diff
@@ -0,0 +1,174 @@
+  };
+
+  it("renders pref rollout forms", async () => {
+    setup();
+    const { getAllByText, container } = await render(
+      <DesignForm experimentType={"rollout"} />,
+    );
+
+    await waitForFormToLoad(container);
+    expect(Api.makeApiRequest).toHaveBeenCalled();
+
+    expect(getAllByText("Description")).toHaveLength(1);
+    expect(getAllByText("Pref Branch")).toHaveLength(1);
+  });
+
+  it("click on pref radio button switches to pref forms", async () => {
+    const mockSuccessResponse = PrefRolloutFactory.build();
+    mockSuccessResponse.rollout_type = "addon";
+    jest
+      .spyOn(Api, "makeApiRequest")
+      .mockImplementation(() => Promise.resolve(mockSuccessResponse));
+    const designForm = await render(<DesignForm experimentType={"rollout"} />);
+    const { getByText, getByLabelText, container } = designForm;
+
+    await waitForFormToLoad(container);
+
+    expect(Api.makeApiRequest).toHaveBeenCalled();
+    expect(getByText("Signed Add-On URL")).toBeInTheDocument();
+
+    fireEvent.click(getByLabelText("Pref Rollout"));
+
+    expect(getByText("Pref Value")).toBeInTheDocument();
+    expect(getByText("Pref Name")).toBeInTheDocument();
+    expect(getByText("Pref Type")).toBeInTheDocument();
+  });
+
+  it("Saves and Redirects", async () => {
+    delete location.replace;
+    location.replace = jest.fn();
+    setup();
+    const { getByText, container } = await render(
+      <DesignForm experimentType={"rollout"} />,
+    );
+    await waitForFormToLoad(container);
+    fireEvent.click(getByText("Save Draft and Continue"));
+    expect(getByText("Save Draft and Continue")).toHaveAttribute("disabled");
+
+    expect(Api.makeApiRequest).toHaveBeenCalledTimes(2);
+    let button = getByText("Save Draft and Continue");
+    // wait for button to no longer be disabled
+    await waitForDomChange({ button });
+    expect(location.replace).toHaveBeenCalled();
+  });
+
+  it("Cancels and nothing is saved ", async () => {
+    setup();
+    const { getByText, container } = await render(
+      <DesignForm experimentType={"rollout"} />,
+    );
+
+    await waitForFormToLoad(container);
+
+    fireEvent.click(getByText("Cancel Editing"));
+
+    expect(Api.makeApiRequest).toHaveBeenCalledTimes(1);
+  });
+
+  it("Make Edits to Form and is saved ", async () => {
+    const data = setup();
+    const { getByTestId, getByText, container } = await render(
+      <DesignForm experimentType={"rollout"} />,
+    );
+
+    await waitForFormToLoad(container);
+    expect(Api.makeApiRequest).toHaveBeenCalledTimes(1);
+
+    const design = "it's my design description value";
+    const prefType = "boolean";
+    const prefName = "browser.enabled";
+    const prefValue = "true";
+
+    const designInput = getByTestId("design");
+    const prefTypeInput = getByTestId("prefType");
+    const prefNameInput = getByTestId("prefName");
+    const prefValueInput = getByTestId("prefValue");
+
```

Comment:
You could use getByLabelText instead of getByTestId right? If you need to, due to other text like "required" being in the label, you can pass an expression, then you wouldn't have to add that prop to your DesignInput. Like this:

`const prefNameInput = getByLabelText(/Pref Name/)`

Reply:
Oh good idea!!  Yeah I don't like adding these test specific ids, feels unnecessary.